### PR TITLE
Supply shuttle are now faction based

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Prework for HvH mode, more support for a factionned req system.

Two supply shuttles, one for each faction.

Tested with loyalist faction, should not affect normal gamemode

## Changelog
:cl:
code: More support for faction in req
/:cl:

### CODE REVIEW

File: _maps/shuttles/supply_rebel.dmm [hunk 1]
```diff
@@ -0,0 +1,47 @@
+//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE
```
Comment (BraveMole) [inline _maps/shuttles/supply_rebel.dmm:1]: Copy pasta much, but shuttle system still doesn't allow to have the same shuttle with different ids

Comment (TiviPlus) [reply _maps/shuttles/supply_rebel.dmm:1]: Then fix it you shouldnt have to spam new shuttles every time

File: code/modules/reqs/supply.dm [hunk 2]
```diff
@@ -441,11 +480,11 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
...
 				else
 					playsound(SSshuttle.supply.return_center_turf(), 'sound/machines/elevator_move.ogg', 50, 0)
-					SSshuttle.moveShuttle("supply", "supply_away", TRUE)
+					SSshuttle.moveShuttle(shuttle_id, away_id, TRUE)
 			else
-				var/obj/docking_port/D = SSshuttle.getDock("supply_home")
+				var/obj/docking_port/D = SSshuttle.getDock(home_id)
 				playsound(D.return_center_turf(), 'sound/machines/elevator_move.ogg', 50, 0)
-				SSshuttle.moveShuttle("supply", "supply_home", TRUE)
+				SSshuttle.moveShuttle(shuttle_id, home_id, TRUE)
```
Comment (TiviPlus) [inline code/modules/reqs/supply.dm:487]: will away require additional mapping because it should be sending to transitspace instead

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/reqs/supply.dm (hunk 2 above)
Comment (BraveMole) [reply code/modules/reqs/supply.dm:470]: fixed

File: _maps/shuttles/supply_rebel.dmm (hunk 1 above)
Comment (BraveMole) [reply _maps/shuttles/supply_rebel.dmm:1]: fixed

**Review by TiviPlus — APPROVED** [review]
tested?

**Comment by BraveMole** [comment]
yes.

Could us a TM though maybe, but i've tested on normal map (POS) and on twins pillars

**Comment by TiviPlus** [comment]
ok

**Comment by TiviPlus** [comment]
No issues

**Pull request merged**